src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/StrandArtifact.java

```diff
@@ -0,0 +1,178 @@
+        }
+
+        unnormalized_posterior_probabilities[StrandArtifactZ.ART_FWD.ordinal()] = pi[StrandArtifactZ.ART_FWD.ordinal()] * likelihoodForArtifactFwd;
+        unnormalized_posterior_probabilities[StrandArtifactZ.ART_REV.ordinal()] = pi[StrandArtifactZ.ART_REV.ordinal()] * likelihoodForArtifactRev;
+
+
+        // find the maximum a posteriori estimate for allele fraction given strand artifact
+        final int maxAlleleFractionIndexFwd = MathUtils.maxElementIndex(forwardLikelihoodsOfAlleleFractions);
+        final int maxAlleleFractionIndexRev = MathUtils.maxElementIndex(reverseLikelihoodsOfAlleleFractions);
+        maximum_a_posteriori_allele_fraction_estimates[StrandArtifactZ.ART_FWD.ordinal()] = gaussIntegrationAbscissasForAlleleFraction.get(maxAlleleFractionIndexFwd);
+        maximum_a_posteriori_allele_fraction_estimates[StrandArtifactZ.ART_REV.ordinal()] = gaussIntegrationAbscissasForAlleleFraction.get(maxAlleleFractionIndexRev);
+
+        // Compute the posterior probability of NO_ARTIFACT; evaluate a single integral over the allele fraction
+        // TODO: write a dot product function and add to MathUtils
+        final List<Double> integrandsForNoArtifact = gaussIntegrationAbscissasForAlleleFraction.stream()
+                .map(f -> getIntegrandGivenNoArtifact(f, numReadsForward, numReadsReverse, numAltReadsForward, numAltReadsReverse))
+                .collect(Collectors.toList());
+        final double likelihoodForNoArtifact = IntStream.range(0, numIntegPointsForAlleleFraction).
+                mapToDouble(i -> gaussIntegrationWeightsForAlleleFraction.get(i) * integrandsForNoArtifact.get(i)).sum();
+
+        unnormalized_posterior_probabilities[StrandArtifactZ.NO_ARTIFACT.ordinal()] = pi[StrandArtifactZ.NO_ARTIFACT.ordinal()] * likelihoodForNoArtifact;
+        // if z = NO_ARTIFACT, MAP estimate for f is the sample alt allele fraction
+        maximum_a_posteriori_allele_fraction_estimates[StrandArtifactZ.NO_ARTIFACT.ordinal()] = -1.0; // TODO: get the allele fraction from the variant context
+
+        final double[] posterior_probabilities = MathUtils.normalizeFromRealSpace(unnormalized_posterior_probabilities);
+
+        gb.attribute(POSTERIOR_PROBABILITIES_KEY, posterior_probabilities);
+        gb.attribute(MAP_ALLELE_FRACTIONS_KEY, maximum_a_posteriori_allele_fraction_estimates);
+    }
+
+    @Override
+    public List<VCFFormatHeaderLine> getDescriptions() {
+        return Arrays.asList(new VCFFormatHeaderLine(POSTERIOR_PROBABILITIES_KEY, 3, VCFHeaderLineType.Float, "posterior probabilities of the presence of strand artifact"),
+                new VCFFormatHeaderLine(MAP_ALLELE_FRACTIONS_KEY, 3, VCFHeaderLineType.Float, "MAP estimates of allele fraction given z"));
+    }
+
+
+    private double getIntegrandGivenNoArtifact(final double f, final int nPlus, final int nMinus, final int xPlus, final int xMinus){
+        final BinomialDistribution binomFwd = new BinomialDistribution(nPlus, f);
```

Comment:
Done. Now I use `new BetaDistribution(null, ALPHA, BETA);` and `MathUtils.binomialProbability`